test/k8sT/Policies.go

```diff
@@ -376,6 +379,81 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			Expect(err).Should(HaveOccurred())
 		})
+
+		It("Denies traffic with k8s default-deny policy", func() {
+			namespace := "default"
+
+			kubectl.Apply(demoPath).ExpectSuccess("Creating demo app")
+			defer kubectl.Delete(demoPath).ExpectSuccess()
+
+			ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil())
+
+			By("Testing connectivity without any policy loaded")
+			clusterIP, _, err := kubectl.GetServiceHostPort(namespace, "app1-service")
+			Expect(err).To(BeNil(), "Cannot get service on %q namespace", namespace)
+			appPods := getAppPods(namespace)
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App2],
+				fmt.Sprintf("curl http://%s/public", clusterIP))
+			Expect(err).Should(BeNil())
+
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App3],
+				fmt.Sprintf("curl http://%s/public", clusterIP))
+			Expect(err).Should(BeNil())
+
+			By("Installing ingress default-deny")
+
+			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, denyIngress, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
+			defer kubectl.Delete(denyIngress).ExpectSuccess()
+
+			err = waitUntilEndpointUpdates(ciliumPod, eps, 4)
+			Expect(err).To(BeNil(), "Waiting for endpoint updates on %s", ciliumPod)
+
+			By("Testing connectivity with ingress default-deny policy loaded")
+
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(HaveOccurred())
+
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(HaveOccurred())
+
+			By("Installing egress default-deny")
+
+			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, denyEgress, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
```

Comment:
Instead of saying 'L3 Policy', I'd prefer you reference `denyEgress` here. Same goes for `denyIngress` above. Makes it more explicit which policy is not working in the case of failure.

Reply:
Done